### ISSUE TITLE
Repicked catalog contains events without correlation.

### ISSUE THREAD

**Issue by 19giovi87** [eqcorrscan/EQcorrscan#406]
**Describe the bug**
Events for which repicking has failed for some reasons (es. gappy data) are still present in the catalog and have no cross-correlation value.

**To Reproduce**
Use the template [1478671.zip](https://github.com/eqcorrscan/EQcorrscan/files/4734651/1478671.zip) and scan the day 2010-02-06, use a MAD threshold of 7.5. Obspy client is 'INGV'.
```python
from obspy import UTCDateTime
from obspy.clients.fdsn import Client

from eqcorrscan.core.match_filter import read_tribe

import logging

logging.basicConfig(level="DEBUG")

CLIENT = Client("INGV")

t1 = UTCDateTime(2010, 2, 6)
t2 = UTCDateTime(2010, 2, 7)

THRESHOLD = 7.5

tribe = read_tribe("1478671.tgz")

party, st = tribe.client_detect(
    client=CLIENT, starttime=t1, endtime=t2, 
    threshold=THRESHOLD,threshold_type="MAD", trig_int=2.0, 
    plot=False, return_stream=True, ignore_length=True,
    ignore_bad_data=True, parallel_process=True, process_cores=6, cores=12)
```
A detection will occur at **2010-02-06 - 01:41:36**. If you repick the catalog, this will fail and events will still be present, showing no correlation value.
```python
repicked_catalog = party.lag_calc(
    st, pre_processed=False, shift_len=0.1,
    min_cc=0.6, ignore_bad_data=True,
    return_correlations=True, parallel=True)
```
The associated qml file for this event will look  like:
```
 </pick>
      <pick publicID="smi:local/bcdd1193-4022-49b1-8e31-cc1c16b99e2e">
        <time>
          <value>2010-02-06T01:41:43.119800Z</value>
        </time>
        <waveformID networkCode="IV" stationCode="MSRU" locationCode="" channelCode="HHE"></waveformID>
      </pick>
      <pick publicID="smi:local/dae0a864-1f99-4711-b01e-5442ae9c1d2a">
        <time>
          <value>2010-02-06T01:41:38.959800Z</value>
        </time>
        <waveformID networkCode="IV" stationCode="MSRU" locationCode="" channelCode="HHZ"></waveformID>
      </pick>
      <pick publicID="smi:local/3adf6913-d550-41ef-a73d-f8385071ac64">
        <time>
          <value>2010-02-06T01:41:38.220000Z</value>
        </time>
        <waveformID networkCode="IV" stationCode="MTTG" locationCode="" channelCode="HHE"></waveformID>
      </pick>
```
But should be like:

```
 <pick publicID="smi:local/517c8a5f-0839-49f1-8242-0493f0aee35f">
        <time>
          <value>2010-12-20T06:03:02.860000Z</value>
        </time>
        <waveformID networkCode="IV" stationCode="MPAZ" locationCode="" channelCode="HHE"></waveformID>
        <methodID>smi:local/EQcorrscan</methodID>
        <phaseHint>S</phaseHint>
        <evaluationMode>automatic</evaluationMode>
        <comment id="smi:local/11cc287d-1a78-4c31-9e97-ac1d50d69019">
          <text>cc_max=0.6666431427001953</text>
        </comment>
        <creationInfo>
          <agencyID>eqcorrscan.core.lag_calc</agencyID>
        </creationInfo>
      </pick>
      <pick publicID="smi:local/7219a32b-b56a-4347-bce9-a8c82c1a57b6">
        <time>
          <value>2010-12-20T06:02:59.400000Z</value>
        </time>
        <waveformID networkCode="IV" stationCode="MPNC" locationCode="" channelCode="HHZ"></waveformID>
        <methodID>smi:local/EQcorrscan</methodID>
        <phaseHint>P</phaseHint>
        <evaluationMode>automatic</evaluationMode>
        <comment id="smi:local/ba187327-cd8f-4560-af76-0a2cede27a09">
          <text>cc_max=0.6770036816596985</text>
        </comment>
        <creationInfo>
          <agencyID>eqcorrscan.core.lag_calc</agencyID>
        </creationInfo>
      </pick>
```

**Expected behavior**
The day 2010-02-06 contains many gaps in data and this is probably not handled well by eqcorrscan. When creating the party, if t1 and t2 are 1-hour around the expected detection, everything works smooth, proving that gaps are problematic even when these are distant from the detection.

**Desktop (please complete the following information):**
 - Operating System: mac os
 - Python version:      3.6
 - EQcorrscan version: 0.4.1

**Comment by calum-chamberlain** [comment]
Edited to have all the code required for reproduction.

@19giovi87 and I have had an email exchange about this. I can reproduce but haven't had time to fix this, and don't know when I will have time.  This boils down to two issues I think:
1. The returned event contains the picks assigned by the template, ideally you don't want that, if an event isn't repicked it shouldn't be in the repicked catalogue! You can work around this at the moment by checking if those picks have comments that contain a correlation value. This should be simple to fix in EQcorrscan at some point.
2. When I set `t1` and `t2` to the hour around the detection then all works well. It looks like you are working with some really gappy data and EQcorrscan is not handling this well.  This might be a little more difficult to fix, but should be done.

I think the first issue is fairly simple to fix: `None` should be returned when an event is not repicked. The second issue requires a bit more tracking down. I'm not sure why the `detect` is allowing these data, but `lag_calc` is not.

**Comment by calum-chamberlain** [comment]
Okay, so it looks like it is the `ignore_length` argument that is the main issue here - `.lag_calc` doesn't have an `ignore_length` argument, so these data were allowed by `.detect` which does have an `ignore_length`, but not by `.lag_calc`. The fix is to add an `ignore_length` argument. 

I'm just running some tests and will open a PR shortly with this fix. I'm hoping to get out a new release in the next few days which will have this patch.

**Comment by calum-chamberlain** [comment]
Closed by #414
